### ISSUE TITLE
move_up_view/move_down_view issue caused by _tree_manager filtering

### ISSUE THREAD

**Issue by smcoll** [edoburu/django-polymorphic-tree#37]
The move up/down arrows in the admin are only working for me when the objects sharing the same level and parent are also the same node type.

This appears to be because the `get_previous_sibling()`/`get_next_sibling()` methods from django-mptt are returning querysets only containing the same types of nodes as the original.  Using `_mptt_filter` instead of `filter` at https://github.com/django-mptt/django-mptt/blob/0.7.2/mptt/models.py#L615 seems to solve the problem.

Perhaps this is an upstream issue (i think i'll post something there) but i don't know if they'll have a compelling reason to make the change.

Django==1.7.7
django-mptt==0.7.2
django-polymorphic==0.7
django-polymorphic-tree==1.1

**Comment by smcoll** [comment]
django-mptt marked my ticket as "wontfix", so we'll need to work out a resolution on this end.

**Comment by smcoll** [comment]
Here's an example of what i mean.  `get_siblings()` returns an object, but `get_next_sibling()` nor `get_previous_sibling()` do not.

```
>>> n = BaseTreeNode.objects.get(id=26)
>>> n.get_siblings()
[<Foo: foo>]
>>> n.get_next_sibling()
>>> n.get_previous_sibling()
>>>
```

**Comment by smcoll** [comment]
`_tree_manager.filter()` doesn't seem to return any results:

```
>>> n._tree_manager.filter()
>>> n._tree_manager.all()
[<Foo: foo>, <Foo: foo2>]
>>> n._tree_manager._mptt_filter()
[<Foo: foo>, <Foo: foo2>, <Bar: bar>, <Bar: bar2>, ...]
```

**Comment by smcoll** [comment]
Getting closer: for whatever reason, when `filter()` or `all()` is called on `n._tree_manager`, the queryset is instantiated with the child model as an argument.  When `n._tree_manager._mptt_filter()` is called, the queryset is instantiated with the base model as an argument.

**Comment by vdboor** [comment]
This is fixed in 1.2

The issue was that the `_mptt_filter()` received a queryset of a child model, so it didn't find all MPTT objects that the base model links to. We patch the queryset to point to the base model again - which fixes this issue!
